src/query.rs

```diff
@@ -18,4 +19,37 @@ impl Query {
     fn __repr__(&self) -> PyResult<String> {
         Ok(format!("Query({:?})", self.get()))
     }
+
+    /// Construct a Tantivy's TermQuery
+    #[staticmethod]
+    #[pyo3(signature = (schema, field_name, field_value, index_option = "position"))]
+    pub(crate) fn new_term_query(
+        schema: &Schema,
+        field_name: &str,
+        field_value: &PyAny,
+        index_option: &str,
+    ) -> PyResult<Query> {
+        make_term_query(schema, field_name, field_value, index_option)
```

Comment:
Why the separate function to implement this instead of directly here in the method?

Reply:
It's just for readability. I suppose we will have several factory methods like this for other query types.
But having all implementations here is also fine.

Reply:
I moved the implementation to `term_query()` instead of using a utility function. https://github.com/quickwit-oss/tantivy-py/pull/175/commits/68bc4668e5f86cdcb47bdcfcac31c2812bd1d50f